[Cases] Wrap the all cases list tags column

adcoelho opened the pull request:
## Summary

This PR aims to stop the truncating behavior from the Tags column in the all-cases list.

* An ellipsis shows up if there are too many tags
* The height is limited to three rows
* The tooltip behavior remains
* Fixed the tags column width

### Screenshots

<details><summary>Before:</summary>
<img width="1408" alt="Screenshot 2022-11-08 at 10 48 34" src="https://user-images.githubusercontent.com/1533137/200531920-03442218-7151-471b-9ff7-66f31acf1f56.png">
</details>

<details><summary>After</summary>
<img width="1425" alt="Screenshot 2022-11-08 at 10 47 51" src="https://user-images.githubusercontent.com/1533137/200531952-f2c9be6e-0a7d-41e2-bd8c-03c6f36635d0.png">
</details>

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

elasticmachine commented:
Pinging @elastic/response-ops-cases (Feature:Cases)

jonathan-buttner commented:
Looks like we're overflowing into the next column when the tag is really long

![image](https://user-images.githubusercontent.com/56361221/200606392-d913e386-1b4e-4be2-a0cf-449fab7322d1.png)

peteharverson commented on x-pack/plugins/cases/public/components/all_cases/use_cases_columns.tsx:
```diff
@@ -58,6 +59,16 @@ const MediumShadeText = styled.p`
 `;
 
+const LINE_CLAMP = 3;
+const LineClampedEuiBadgeGroup = euiStyled(EuiBadgeGroup)`
```
As discussed, depending on the length of the tags, the ellipsis sometimes isn't displayed and the tooltip doesn't show more than 3 lines so additional tags are not shown in the tooltip.

For example here I have 9 tags added to the case:

<img width="422" alt="image" src="https://user-images.githubusercontent.com/7405507/200819015-302be9d6-f78c-46bc-b84b-b3f6d4024de5.png">

but only 6 are shown in the cases list tooltip:

<img width="532" alt="image" src="https://user-images.githubusercontent.com/7405507/200819170-af0982bf-04ea-40cf-a720-5c8dbcf38bdd.png">

peteharverson commented:
> Looks like we're overflowing into the next column when the tag is really long

Maybe the width of the badge itself could be limited, as happens in the sidebar of the case view page

<img width="190" alt="image" src="https://user-images.githubusercontent.com/7405507/200819470-92900cc0-a7f5-41e9-b04b-9f10bcd3ff23.png">

adcoelho commented:
Updated this PR to fix your findings @peteharverson @jonathan-buttner 

Below you can see how the tooltip is not restricted to 3 rows anymore.

<img width="696" alt="Screenshot 2022-11-09 at 16 11 37" src="https://user-images.githubusercontent.com/1533137/200867696-0fbc4be1-d888-49e3-92ff-ba887ed8a8f4.png">

I also fixed the length of the badges so that the ellipsis is triggered in the main table view.
(The tooltip ellipsis is the old behaviour.)

<img width="776" alt="Screenshot 2022-11-09 at 16 16 51" src="https://user-images.githubusercontent.com/1533137/200868696-c96aec1a-8da8-4e27-9c94-45b89224de20.png">

I also fixed the width of the alerts column as discussed offline.

adcoelho commented:
@elasticmachine merge upstream

peteharverson reviewed (approved):
Tested and LGTM

jonathan-buttner reviewed (approved):
Looks good!
